core/src/services/s3/config.rs

```diff
@@ -196,6 +196,13 @@ pub struct S3Config {
 
     /// Indicates whether the client agrees to pay for the requests made to the S3 bucket.
     pub enable_request_payer: bool,
+
+    /// Enable Content-MD5 header for upload operations.
+    ///
+    /// When enabled, OpenDAL will calculate and include the Content-MD5 header
+    /// for PUT operations (single uploads, multipart parts, and append operations).
+    /// This header provides an additional layer of data integrity verification.
+    pub enable_content_md5: bool,
```

Comment:
I don't think it's the correct approch to implement data integrity verification.

Maybe we can take a look over https://github.com/apache/opendal/issues/5549

Reply:
How about implementing `content_md5` as one of `checksum_algorithm`?

Reply:
sounds good.